docs/book/src/developer/repository-layout.md

```diff
@@ -1,3 +1,175 @@
+
+[~/controllers](https://github.com/kubernetes-sigs/cluster-api/tree/main/controllers)
+
+This folder contains reconciler types which provides access to CAPI controllers present in [~/internal/controllers](https://github.com/kubernetes-sigs/cluster-api/tree/main/internal/controllers) directory to our users. These types can be used users to provide a single binary "batteries included" with Cluster API and providers of choice.
+
+### Documentation
+
+[~/docs](https://github.com/kubernetes-sigs/cluster-api/tree/main/docs)
+
+This folder is a place for proposals, developer release guidelines and the Cluster API book.
+
+### Tools
+
+[~/hack](https://github.com/kubernetes-sigs/cluster-api/tree/main/hack)
+
+This folder has scripts used for building, testing and developer workflow. Its subfolder [~/hack/observability](https://github.com/kubernetes-sigs/cluster-api/tree/main/hack/observability) is a place for resources related to  observability like Loki, graphing,  prometheus and promtail.
+
+[~/scripts](https://github.com/kubernetes-sigs/cluster-api/tree/main/scripts)
+
+This folder consists of CI scripts related to setup, build and e2e tests. These are mostly called by CI jobs.
+
+### CAPI as a library
+
+[~/util](https://github.com/kubernetes-sigs/cluster-api/tree/main/util)
+
+This folder contains utilities which are meant to be  used in multiple CAPI modules, providers and users of CAPI.
+
+[~/feature](https://github.com/kubernetes-sigs/cluster-api/tree/main/feature)
+
+This package provides feature gate management used in Cluster API as well as providers. This implementation of feature gates is shared across all providers.
+
+[~/errors](https://github.com/kubernetes-sigs/cluster-api/tree/main/errors)
+
+This is a place for defining errors returned by CAPI. Error types defined here can be used by users of CAPI and the providers.
+
+### Experimental features
+
+[~/exp](https://github.com/kubernetes-sigs/cluster-api/tree/main/exp)
+
+This folder contains experimental features of CAPI. Experimental features are unreliable until they are promoted to the main repository. Each experimental feature is supposed to be present in a subfolder of [~/exp](https://github.com/kubernetes-sigs/cluster-api/tree/main/exp) folder. For e.g. ClusterResourceSet is present inside [~/exp/addons](https://github.com/kubernetes-sigs/cluster-api/tree/main/exp/addons) folder. Historically, machine pool resources are not present in a sub-directory. Migrating them to a subfolder like `~/exp/machinepools` is still pending as it can potentially break existing users who are relying on existing folder structure.
+
+These experimental features are kubebuilder projects having standard folder structure except that the CRDs for these experimental features are present in [config](#configuration) folder of the main repository. These CRDs are deployed along with CAPI cluster. However, these features can be enabled and disabled using feature gates. 
```

Comment:
```suggestion
These features can be enabled and disabled using feature gates supplied to the core Cluster API controller.
```

Reply:
CRDs for experimental features are present outside `exp` directory in `config` folder. Also, these CRDs are deployed in the cluster irrespective of the feature gate value.
This information might be useful so can we include it here?

Reply:
I can definitely see the argument for including it. I have no problem keeping it here.